src/lib/Libifl/pbs_loadconf.c

```diff
@@ -308,12 +308,8 @@ parse_psi(char *conf_value)
         			get_fullhostname(pbs_conf.psi[i].name, pbs_conf.psi[i].name, PBS_MAXHOSTNAME);
 		}
 
-		if (pbs_conf.psi[i].port == 0) {
-			if (is_same_host(pbs_conf.psi[i].name, pbs_conf.pbs_server_name))
-				pbs_conf.psi[i].port = pbs_conf.batch_service_port;
-			else
-				pbs_conf.psi[i].port = PBS_BATCH_SERVICE_PORT;
-		}
+		if (pbs_conf.psi[i].port == 0)
+			pbs_conf.psi[i].port = PBS_BATCH_SERVICE_PORT;
```

Comment:
Can you explain this change to me. I do not follow. Why you removed the if block?

Reply:
The problem with the old code is this: when a server doesn't have port number in PSI, each server sets the port number to  pbs_conf.batch_service_port. So, let's say **PBS_SERVER_INSTANCES=pbs, pbs-2:18002** then, pbs_conf.batch_service_port for server 1 will be set to 15001 and pbs_conf.batch_service_port for server 2 will be set to 18002. So, when server 2 reads PSI, it will see that 'pbs' has no port number, so it will set it to its pbs_conf.batch_service_port, which will be 18002, not 15001.

In general, when a port number is absent, we should assume the default of 15001, that's why I changed it to do just that. Does it make sense?

Reply:
It does. Thanks.